tree-sitter/src/TreeSitter/Symbol.hs

```diff
@@ -1,95 +1,133 @@
-{-# LANGUAGE DeriveLift, ScopedTypeVariables, LambdaCase #-}
+{-# LANGUAGE DeriveLift, LambdaCase, ScopedTypeVariables #-}
+
 module TreeSitter.Symbol
-( TSSymbol
-, fromTSSymbol
-, SymbolType(..)
-, Symbol(..)
-, symbolToName
-, escapeOperatorPunctuation
-) where
+  ( TSSymbol
+  , fromTSSymbol
+  , SymbolType(..)
+  , Symbol(..)
+  , symbolToName
+  , toHaskellCamelCaseIdentifier
+  , toHaskellPascalCaseIdentifier
```

Comment:
I moved all the functions that are handling the mapping of names in here, along with a couple of string helpers that are now only used directly in the tests.